### DISCUSSION TITLE
Why prepareRow

### DISCUSSION THREAD

**Discussion by maraisr** [TanStack/table#2467] [General]
Why do we _have_ to call `prepareRow` on the rows?

Why could this not have been done for us from the columns? I feel like if I understand why, I might be able to leverage this better!

---
I don't think this has been asked/answered before. Please let me know if it has!

**Comment by gargroh** [comment]
This has been done to delegate costly row operations...

**Comment by Fullstack-WEB-Blockchain-Developer** [comment]
> Why could this not have been done for us from the columns? I feel like if I understand why, I might be able to leverage this better!

@maraisr  I have the same issue.
If you find an answer to this, please let me know.
